lib/fileset/default.nix

```diff
@@ -148,6 +153,32 @@ let
 
 in {
 
+  /*
+    Create a file set from a path that may or may not exist:
+    - If the path does exist, the path is [coerced to a file set](#sec-fileset-path-coercion).
+    - If the path does not exist, a file set containing no files is returned.
+
+    Type:
+      maybeMissing :: Path -> FileSet
+
+    Example:
+      # All files in the current directory, but excluding main.o if it exists
+      difference ./. (maybeMissing ./main.o)
+  */
+  maybeMissing =
+    path:
+    if ! isPath path then
+      if isStringLike path then
+        throw ''
+          lib.fileset.maybeMissing: Argument ("${toString path}") is a string-like value, but it should be a path instead, see https://nixos.org/manual/nix/stable/language/values#type-path.''
+      else
+        throw ''
+          lib.fileset.maybeMissing: Argument is of type ${typeOf path}, but it should be a path instead, see https://nixos.org/manual/nix/stable/language/values#type-path.''
```

Comment:
I don't think we need links to the manual. It's spammy and doesn't exactly make things easier with maintaining that web stuff in the long run.

Reply:
Fair enough, removed them again!